core/src/main/scala/org/apache/spark/storage/BlockManagerMaster.scala

```diff
@@ -223,7 +227,18 @@ class BlockManagerMaster(
         log"${MDC(ERROR, e.getMessage)}", e)
     )(ThreadUtils.sameThread)
     if (blocking) {
-      // the underlying Futures will timeout anyway, so it's safe to use infinite timeout here
+      handleRemoveBlockBlockingTimeout(future)
+    }
+  }
+
+  private def handleRemoveBlockBlockingTimeout(future: Future[_]): Unit = {
+    if (cleanBlockBlockingTimeout.isDefined) {
+      new RpcTimeout(FiniteDuration(cleanBlockBlockingTimeout.get, TimeUnit.SECONDS),
+        CLEANER_REFERENCE_TRACKING_BLOCKING_TIMEOUT.key).awaitResult(future)
+    } else {
+      // Normally, the underlying Futures will timeout anyway,
+      // so it's safe to use infinite timeout here. In extreme case,
+      // Driver can't crease thread handling this rpc, here will be stuck forever.
       RpcUtils.INFINITE_TIMEOUT.awaitResult(future)
```

Comment:
Do we still want this case? It look like a bug to me.

Reply:
Yea，so what do you think is a good default value for CLEANER_REFERENCE_TRACKING_BLOCKING_TIMEOUT?